test/unit/NodeCompiler/CompileNodeToValueTest.php

```diff
@@ -741,6 +741,29 @@ class Bat {
         $classInfo->getConstant('ONE_VALUE')->getValue();
     }
 
+    public function testEnumValueThrowsException(): void
+    {
+        $phpCode = <<<'PHP'
+        <?php
+
+        enum Foo: int {
+            case ONE = 1;
+        }
+        class Bat {
+            const ONE_VALUE = Foo::ONE;
+        }
+        PHP;
+
+        $reflector = new DefaultReflector(new AggregateSourceLocator([
+            new StringSourceLocator($phpCode, $this->astLocator),
+            new PhpInternalSourceLocator($this->astLocator, $this->sourceStubber),
+        ]));
+        $classInfo = $reflector->reflectClass('Bat');
+
+        $this->expectException(UnableToCompileNode::class);
```

Comment:
We should check at least part of the exception message in this test: it seems like the mutation test reports no behavioral change (removes `throw` and is happy with that), which indicates that an assertion is to be added here